New Resource: `azurerm_workloads_sap_discovery_virtual_instance` (#24342)
hashicorp/terraform-provider-azurerm: website/docs/r/workloads_sap_discovery_virtual_instance.html.markdown

---
subcategory: "Workloads"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_workloads_sap_discovery_virtual_instance"
description: |-

>   Manages a SAP Discovery Virtual Instance.

Author: Updated

---

# azurerm_workloads_sap_discovery_virtual_instance

> Manages a SAP Discovery Virtual Instance.

Author: Updated

-> **Note:** Before using this resource, it's required to submit the request of registering the Resource Provider with Azure CLI `az provider register --namespace "Microsoft.Workloads"`. The Resource Provider can take a while to register, you can check the status by running `az provider show --namespace "Microsoft.Workloads" --query "registrationState"`. Once this outputs "Registered" the Resource Provider is available for use.

## Example Usage

```hcl
data "azurerm_subscription" "current" {}

resource "tls_private_key" "example" {
  algorithm = "RSA"
  rsa_bits  = 4096
}

data "tls_public_key" "example" {
  private_key_pem = tls_private_key.example.private_key_pem

> }

Reviewer: shall we remove 3rd party resources like `tls_private_key` and `tls_public_key`?

Author: Updated

        size_in_gb      = 256
        sku_name        = "Premium_LRS"
      }

      disk_volume_configuration {
        volume_name     = "usr/sap"
        number_of_disks = 1
        size_in_gb      = 128
        sku_name        = "Premium_LRS"
      }

      disk_volume_configuration {
        volume_name     = "backup"
        number_of_disks = 2
        size_in_gb      = 256
        sku_name        = "StandardSSD_LRS"
      }

      disk_volume_configuration {
        volume_name     = "os"
        number_of_disks = 1
        size_in_gb      = 64
        sku_name        = "StandardSSD_LRS"
      }

      virtual_machine_full_resource_names {
        host_name               = "apphostName0"
        os_disk_name            = "app0osdisk"
        virtual_machine_name    = "appvm0"
        network_interface_names = ["appnic0"]

        data_disk_names = {
          default = "app0disk0"
        }
      }
    }
  }

  identity {
    type = "UserAssigned"

    identity_ids = [
      azurerm_user_assigned_identity.example.id,
    ]
  }

  depends_on = [
    azurerm_role_assignment.example
  ]
}
```

## Arguments Reference

The following arguments are supported:

> * `name` - (Required) The name which should be used for this SAP Discovery Virtual Instance. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `name` - (Required) Specifies the name of the SAP Discovery Virtual Instance. Changing this forces a new resource to be created.
```

Author: Updated

* `central_server_virtual_machine_id` - (Required) The ID of the Virtual Machine of the Central Server. Changing this forces a new resource to be created.

* `environment` - (Required) The environment type for the SAP Discovery Virtual Instance. Possible values are `NonProd` and `Prod`. Changing this forces a new resource to be created.

* `sap_product` - (Required) The SAP Product type for the SAP Discovery Virtual Instance. Possible values are `ECC`, `Other` and `S4HANA`. Changing this forces a new resource to be created.

* `identity` - (Optional) An `identity` block as defined below.

* `managed_resource_group_name` - (Optional) The name of the managed Resource Group for the SAP Discovery Virtual Instance. Changing this forces a new resource to be created.

* `managed_storage_account_name` - (Optional) The name of the custom Storage Account created by the service in the managed Resource Group. Changing this forces a new resource to be created.

* `tags` - (Optional) A mapping of tags which should be assigned to the SAP Discovery Virtual Instance.

---

An `identity` block supports the following:

> * `type` - (Required) The type of Managed Service Identity that should be configured on this SAP Discovery Virtual Instance. Only possible value is `UserAssigned`.

Author: Updated
